### ISSUE TITLE
Code Editor 2: Multiple tab characters in a row leaves caret short by 3 positions when "Tab Type" Preference is set to spaces

### ISSUE THREAD

**Issue by gnysek** [YoYoGames/GameMaker-Bugs#6203]
### Description
In case there's an existing project, which was using `\t` (Tab) for indenting, and Code Editor 2 is set to use `Space` as "Tab Type", it enters wrong number of spaces - it's equal to `tabs number - 3`.

So, for example, when "Tab size" = 4, and there are `\t\t` in current line, instead of adding 8 spaces after pressing enter, it adds 5. Same, if there's `\t\t\t`, it adds 9 instead of 12 spaces.

Seems it works when "Tab Type" is set to "Tab".

### Steps To Reproduce
1. Open/Create script.
2. Put code, which contains `\t` as indenting characters, not spaces, for example:
```gml
function Script1() {
	var test = 1;
	if (test == 1) {
		var test = 2; // press ENTER after this comment
	}
}
```
3. Ensure that "Tab Type" setting for CE2 is set to Spaces
3. Press enter in most indented line
4. Observe that it doesn't add as much characters as expected, and there will be missing 3 spaces
![obraz](https://github.com/YoYoGames/GameMaker-Bugs/assets/524094/3fc59510-079e-419c-a285-cc27a6762f17)

_Update:_
It seems to break same way, if spaces are used instead of `\t`:
![obraz](https://github.com/YoYoGames/GameMaker-Bugs/assets/524094/dc7310fc-8a81-4ef4-8eff-330644d7e216)

### Which version of GameMaker are you reporting this issue for?
IDE v2024.600.0.571 Runtime v2024.600.0.597

### Which operating system(s) are you seeing the problem on?
Windows 10.0.22631.0

[7a9f366a-e0bd-4767-aa42-81157a4ee902](https://api.gamemaker.io/api/github/downloads/7a9f366a-e0bd-4767-aa42-81157a4ee902)

**Comment by JhOnNeY** [comment]
I can confirm the same using
MacOS Sonoma v14.5
IDE v2024.600.0.571 Runtime v2024.600.0.597

Tabbed comments and code does not line up correctly

**Comment by zreedy** [comment]
Fixed.

**Comment by caitlinrmcintyre** [comment]
Verified fix in Beta IDE v2024.800.0.593  Runtime v2024.800.0.617, thanks.
